File: HFU Studiportal/Exam.cpp
//
//  Exam.cpp
//  HFU Studiportal
//
//  Created by Christian Würthner on 15.01.15.
//  Copyright (c) 2015 Hochschule Furtwangen University. All rights reserved.
//

#include "Exam.h"

bool Exam::isResignated() {
    return this->getResignation() == "1";
    
}

Kind Exam::getKindEnum() {
    if(this->getKind() == "PL") {
        return KIND_PL;
        
    } else if(this->getKind() == "VL") {
        return KIND_VL;
        
    } else if(this->getKind() == "P") {
        return KIND_P;
        
    } else if(this->getKind() == "G") {
        return KIND_G;
        
    } else if(this->getKind() == "KO") {
        return KIND_KO;
        
    } else {
        return KIND_UNDEFINED;
        
    }
}

State Exam::getStateEnum() {
    if(this->getState() == "AN") {
        return STATE_AN;
        
    } else if(this->getState() == "BE") {
        return STATE_BE;
        
    } else if(this->getState() == "NB") {
        return STATE_NB;
        
    } else if(this->getState() == "EN") {
        return STATE_EN;
        
    } else {
        return STATE_UNDEFINED;
        
    }
}

Note Exam::getNoteEnum() {
    if(this->getNote() == "GR") {
        return NOTE_GR;
        
    } else if(this->getNote() == "K") {
        return NOTE_K;
        
    } else if(this->getNote() == "SA") {
        return NOTE_SA;
        
    } else if(this->getNote() == "U") {
        return NOTE_U;
        
    } else if(this->getNote() == "VF") {
        return NOTE_VF;
        
    } else {
        return NOTE_UNDEFINED;
        
    }
}

std::string Exam::getStateName() {
    switch (this->getStateEnum()) {
        case STATE_BE: return localizeString("exam.state.be"); break;
        case STATE_AN: return localizeString("exam.state.an"); break;
        case STATE_NB: return localizeString("exam.state.nb"); break;
        case STATE_EN: return localizeString("exam.state.en"); break;
        default: return "Undefined";
            
    }
}

std::string Exam::getNoteName() {
    switch (this->getNoteEnum()) {
        case NOTE_GR: return localizeString("exam.note.gr"); break;
        case NOTE_U: return localizeString("exam.note.u"); break;
        case NOTE_VF: return localizeString("exam.note.vl"); break;
        case NOTE_K: return localizeString("exam.note.k"); break;
        case NOTE_SA: return localizeString("exam.note.sa"); break;
        default: return "Undefined";
            
    }
}

uint32_t Exam::getId() {
    return this->id;
    
}

std::string Exam::getExamNo() {
    return this-> examNo;
    
}

std::string Exam::getName() {
    return this->name;
    
}

std::string Exam::getBonus() {
    return this->bonus;
    
}

std::string Exam::getMalus() {
    return this->malus;
    
}

std::string Exam::getECTS() {
    return this->ects;
    
}

std::string Exam::getSWS() {
    return this->sws;
    
}

std::string Exam::getSemster() {
    return this->semester;
    
}

std::string Exam::getKind() {
    return this->kind;
    
}

std::string Exam::getTryCount() {
    return this->tryCount;
    
}

std::string Exam::getGrade() {
    return this->grade;
    
}

std::string Exam::getState() {
    return this->state;
    
}

std::string Exam::getComment() {
    return this->comment;
    
}

std::string Exam::getResignation() {
    return this->resignation;
    
}

std::string Exam::getNote() {
    return this->note;
    
}

void Exam::setExamNo(std::string value) {
    this->examNo = value;
    
}

void Exam::setName(std::string value) {
    this->name = value;
    
}

void Exam::setBonus(std::string value) {
    this->bonus = value;
    
}

void Exam::setMalus(std::string value) {
    this->malus = value;
    
}

void Exam::setECTS(std::string value) {
    this->ects = value;
    
}

void Exam::setSWS(std::string value) {
    this->sws = value;
    
}

void Exam::setSemester(std::string value) {
    this->semester = value;
    
}

void Exam::setKind(std::string value) {
    this->kind = value;
    
}

void Exam::setTryCount(std::string value) {
    this->tryCount = value;
    
}

void Exam::setGrade(std::string value) {
    this->grade = value;
    
}

void Exam::setState(std::string value) {
    this->state = value;
    
}

void Exam::setComment(std::string value) {
    this->comment = value;
    
}

void Exam::setResignation(std::string value) {
    this->resignation = value;
    
}

void Exam::setNote(std::string value) {
    this->note = value;
    
}

File: HFU Studiportal/Exam.m
//
//  Exam.cpp
//  HFU Studiportal
//
//  Created by Christian Würthner on 15.01.15.
//  Copyright (c) 2015 Hochschule Furtwangen University. All rights reserved.
//

#include "Exam.h"

@implementation Exam

-(id) initWithExamNo:(NSString*) examNo {
    self.examNo = examNo;
    self.name = @"-";
    self.bonus = @"-";
    self.malus = @"-";
    self.ects = @"-";
    self.semester = @"-";
    self.kind = @"-";
    self.tryCount = @"-";
    self.grade = @"-";
    self.state = @"-";
    self.comment = @"-";
    self.resignation = @"-";
    self.note = @"-";
    return self;
    
}

-(BOOL) isResignated {
    return [self.resignation isEqualToString:@"1"];
    
}

-(ExamKind) kindEnum {
    if([[self kind] isEqualToString:@"PL"]) {
        return ExamKindPL;
        
    } else if([[self kind] isEqualToString:@"VL"]) {
        return ExamKindVL;
        
    } else if([[self kind] isEqualToString:@"P"]) {
        return ExamKindP;
        
    } else if([[self kind] isEqualToString:@"G"]) {
        return ExamKindG;
        
    } else if([[self kind] isEqualToString:@"KO"]) {
        return ExamKindKO;
        
    } else {
        return ExamKindUndefined;
        
    }
}

-(ExamState) stateEnum {
    if([[self state] isEqualToString:@"AN"]) {
        return ExamStateAN;
        
    } else if([[self state] isEqualToString:@"BE"]) {
        return ExamStateBE;
        
    } else if([[self state] isEqualToString:@"NB"]) {
        return ExamStateNB;
        
    } else if([[self state] isEqualToString:@"EN"]) {
        return ExamStateEN;
        
    } else {
        return ExamStateUndefined;
        
    }
}

-(ExamNote) noteEnum {
    if([[self note] isEqualToString:@"GR"]) {
        return ExamNoteGR;
        
    } else if([[self note] isEqualToString:@"K"]) {
        return ExamNoteK;
        
    } else if([[self note] isEqualToString:@"SA"]) {
        return ExamNoteSA;
        
    } else if([[self note] isEqualToString:@"U"]) {
        return ExamNoteU;
        
    } else if([[self note] isEqualToString:@"VF"]) {
        return ExamNoteVF;
        
    } else {
        return ExamNoteUndefined;
        
    }
}

-(NSString*) stateName {
    switch ([self stateEnum]) {
        case ExamStateBE: return NSLocalizedString(@"exam.state.be", nil); break;
        case ExamStateAN: return NSLocalizedString(@"exam.state.an", nil); break;
        case ExamStateNB: return NSLocalizedString(@"exam.state.nb", nil); break;
        case ExamStateEN: return NSLocalizedString(@"exam.state.en", nil); break;
        default: return @"Undefined";
            
    }
}

-(NSString*) noteName {
    switch ([self noteEnum]) {
        case ExamNoteGR: return NSLocalizedString(@"exam.note.gr", nil); break;
        case ExamNoteU: return NSLocalizedString(@"exam.note.u", nil); break;
        case ExamNoteVF: return NSLocalizedString(@"exam.note.vl", nil); break;
        case ExamNoteK: return NSLocalizedString(@"exam.note.k", nil); break;
        case ExamNoteSA: return NSLocalizedString(@"exam.note.sa", nil); break;
        default: return @"Undefined";
            
    }
}

@end